### DISCUSSION TITLE
Regarding J++

### DISCUSSION THREAD

**Discussion by Dragas** [microsoft/openjdk#4] [Q&A] [answered]
I remember microsoft already having their own java builds. At that time they called J++. I think they were pulled because of proprietary extensions. What are guarantees that these builds will not turn into another J++ fiasco?

**Comment by jumoog** [comment]
It's not possible to migrate from GPLv2+CE to something proprietary

**Comment by karianna** [comment, marked as answer]
J++ is firmly in the Dustbin of History :-)!  Microsoft Build of OpenJDK is an OpenJDK distribution under the same GPLv2+CE license as all the other OpenJDK vendors such as Azul Zulu, Eclipse Adoptium et al. Microsoft looks forward to continuing its investment and support of OpenJDK as part of that wonderful community!

FYI - This build is an LTS supported release until at least 2024.
